pkg/logs/seek/seek.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package seek
+
+import (
+	"time"
+)
+
+// Strategy represents the stategy a tailer should be using while starting tailing a new element.
+type Strategy int
+
+const (
+	// Start tail from the beginning
+	Start Strategy = iota
+	// Recover tail from a given offset
+	Recover
+	// End tail from the end
+	End
+)
+
+// Registry holds a list of offsets.
+type Registry interface {
+	GetOffset(identifier string) string
+}
+
+// Seeker provides seek strategies to inputs when starting tailing new element.
+type Seeker struct {
+	registry Registry
+}
+
+// NewSeeker returns a new seeker
+func NewSeeker(registry Registry) *Seeker {
+	return &Seeker{
+		registry: registry,
+	}
+}
+
+// origin represents the time reference of the agent.
+var origin = time.Now()
```

Comment:
Let's consider this scenario:
- agent is started
- daylight saving time occurs, shifting current time by -1h
- a new file is created, and tailed by the agent

Are we going to consider the file was created before the agent start, thus tail from the beginning ?

Not sure it's a problem, I'm just trying thinking of the possible pitfalls of storing the start time this way.